### PR TITLE
CON-2824: Fixed an issue that prevented Navigator title updates

### PR DESCRIPTION
**Pull request by kernelkiller**
- replaced the routeWillEnter logic from RoutedConnect by a history
state based approach

### CODE REVIEW

File: components/View/index.jsx [hunk 1]
```diff
@@ -131,19 +140,14 @@ class View extends Component {
...
-    this.props.setTitle(this.props.title);
-  }
-
   /**
    * Renders the HTML meta tags.
    * @returns {JSX}
    */
   renderMetaTags() {
-    const { meta, link, script, style } = this.props.head;
+    const {
```
Comment (richardgorman) [inline components/View/index.jsx]: I think the linter was looking for:

```
const {
  meta,
  link,
  script,
  style,
} = this.props.head;
```

File: components/View/index.jsx [hunk 2]
```diff
@@ -76,13 +80,18 @@ class View extends Component {
...
-    }
-
-    if (nextProps.viewTop && nextProps.viewTop !== this.props.viewTop) {
-      // Scroll to top
-      this.element.scrollTop = 0;
+    // Only run the logic, if the path for the View instance is active
+    if (nextProps.historyPathname === this.pathname) {
+      if ((nextProps.title !== this.props.title) ||
+        (this.props.navigatorTitle !== nextProps.title)) {
+        // Update the title if it distinguishes from the current navigator title
```
Comment (richardgorman) [inline components/View/index.jsx]: `Update the title if it is different from the set navigator title.`

File: components/View/index.jsx [hunk 3]
```diff
@@ -76,13 +80,18 @@ class View extends Component {
...
    */
   componentWillReceiveProps(nextProps) {
-    if (nextProps.title !== this.props.title) {
-      this.props.setTitle(nextProps.title || this.props.title);
-    }
-
-    if (nextProps.viewTop && nextProps.viewTop !== this.props.viewTop) {
-      // Scroll to top
-      this.element.scrollTop = 0;
+    // Only run the logic, if the path for the View instance is active
```
Comment (richardgorman) [inline components/View/index.jsx]: `Only update if the active pathname is the same as the pathname of this route.`

File: components/View/index.jsx [hunk 4]
```diff
@@ -56,6 +58,8 @@ class View extends Component {
   constructor(props) {
     super(props);
 
+    // Store the active pathname at instantiation
```
Comment (richardgorman) [inline components/View/index.jsx]: `// Store the active pathname on instantiation.`

File: components/View/index.jsx [hunk 5]
```diff
@@ -76,13 +80,18 @@ class View extends Component {
...
   componentWillReceiveProps(nextProps) {
-    if (nextProps.title !== this.props.title) {
-      this.props.setTitle(nextProps.title || this.props.title);
-    }
-
-    if (nextProps.viewTop && nextProps.viewTop !== this.props.viewTop) {
-      // Scroll to top
-      this.element.scrollTop = 0;
+    // Only run the logic, if the path for the View instance is active
+    if (nextProps.historyPathname === this.pathname) {
```
Comment (richardgorman) [inline components/View/index.jsx]: The use of a guard clause here would make this block easier to read.

File: components/View/index.jsx (hunk 3 above)
Comment (kernelkiller) [reply components/View/index.jsx]: done

File: components/View/index.jsx (hunk 2 above)
Comment (kernelkiller) [reply components/View/index.jsx]: done

File: components/View/index.jsx (hunk 4 above)
Comment (kernelkiller) [reply components/View/index.jsx]: done

File: components/View/index.jsx (hunk 1 above)
Comment (kernelkiller) [reply components/View/index.jsx]: There is a special option for the related rule, called `multiline` which is inactive.
https://eslint.org/docs/rules/object-curly-newline

So i think it's totally ok with it

File: components/View/index.jsx (hunk 5 above)
Comment (kernelkiller) [reply components/View/index.jsx]: done

**Pull request merged**